WcaOnRails/app/webpacker/components/CompetitionForm/CompsTable.js

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable react/no-danger */
+import React, { useState } from 'react';
+import {
+  Button, Table, Message,
+} from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import Loading from '../Requests/Loading';
+
+function NotConfirmedIcon() {
+  return (
+    <i
+      className="fas fa-exclamation-circle"
+      data-toggle="tooltip"
+      data-placement="top"
+      data-container="body"
+      title="This competition is not confirmed yet."
+    />
+  );
+}
+
+function CompsTableContent({ nearby, action }) {
+  const [showEvents, setShowEvents] = useState(false);
+
+  const headerRow = () => (
```

Comment:
I think something like this could just be a constant instead of a function, unless I'm missing something. Possibly wrapped in a `useMemo` dependent on `showEvents`.

Reply:
Would love to see this as a component just outside of the parent component. It does take a prop "showEvents" so it can't exactly be a constant.

`useMemo` feels like a bit overkill for this simple design.

Reply:
I think I will end up making this a bit more reusable anyway as there was more variation than I had initially planned for in each of the places where its going to be used, so I will reevaluate this later on